core/src/main/java/com/graphhopper/routing/util/OSMParsers.java

```diff
@@ -27,28 +27,43 @@
 import com.graphhopper.storage.IntsRef;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.function.Function;
 
 public class OSMParsers {
+    private final List<String> ignoredHighways;
+    private List<String> acceptedRoutes;
     private final List<TagParser> wayTagParsers;
     private final List<VehicleTagParser> vehicleTagParsers;
     private final List<RelationTagParser> relationTagParsers;
     private final List<RestrictionTagParser> restrictionTagParsers;
     private final EncodedValue.InitializerConfig relConfig = new EncodedValue.InitializerConfig();
 
     public OSMParsers() {
-        this(new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
+        this(new ArrayList<>(), Arrays.asList("ferry", "shuttle_train"), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
```

Comment:
I did another check (for Germany). I counted the number of ways that use a `route` tag, but do **not** use a `highway` tag:

```
ferry: 0.01%, 983
ski: 0.00%, 413
canoe: 0.00%, 241
piste: 0.00%, 193
boat: 0.00%, 37
historic: 0.00%, 26
shuttle_train: 0.00%, 20
bicycle: 0.00%, 9
mudflat_hiking: 0.00%, 8
fitness_trail: 0.00%, 8
hiking: 0.00%, 5
railway: 0.00%, 4
flat: 0.00%, 3
foot: 0.00%, 3
sk:nordic: 0.00%, 3
RB/RE Richtung Rosenheim: 0.00%, 3
nordic: 0.00%, 2
power: 0.00%, 2
pipeline: 0.00%, 2
bus: 0.00%, 1
Horumersiel: 0.00%, 1
water_skiing: 0.00%, 1
fitness-trail: 0.00%, 1
F1 F2 F3: 0.00%, 1
train: 0.00%, 1
19 Haigerloch - Heiligenzimmern; 7402 Sulz; 7402 Horb; 7416 Neckarhausen: 0.00%, 1
yes: 0.00%, 1
tracks: 0.00%, 1
Faulungen / Ort Struth-Steinbruch: 0.00%, 1
harbour_cruise: 0.00%, 1
cruise: 0.00%, 1
Ship: 0.00%, 1
```

So most of them are ferries and everything else occurs only less than 500 times (the total number of accepted ways was >13 million).

So I would say we do not need this list of 'acceptedRoutes' and can simply accept all ways with a 'route' tag. In most cases these will be inaccessible edges in the end, because the vehicle/access parsers do not set the access to true for unknown types of routes(?).